metaflow/client/core.py

```diff
@@ -878,6 +879,72 @@ def tarball(self) -> tarfile.TarFile:
         """
         return self._tar
 
+    @property
+    def extract(self) -> TemporaryDirectory:
```

Comment:
would rename to `extract_user_code` or something like that. We discussed a bit and we could do:
- one method called extract with a flag that determines WHAT (user code, environment, everything)
- `extract_user_code` which would be this and then the tarball extraction would be the other one.

I think we were leaning towards the second solution and that could work but it feels a little less symetric.

I think we also discussed a slightly different UX than the temporary directory. I guess the question is how will users use this.

Reply:
Make it a function instead of a property.

Reply:
Final comment from me on this:
- ok with name `extract`
- make it a function